### PR TITLE
Bugfix GitHub pages deploy

### PR DESCRIPTION
**Pull request by adku1173**
* fixes missing artifacts after download by explicitly specifying download path
* fixes sphinx warnings encountered when docs are build via `hatch -v run docs:build`
* fixes missing autosummary for `acoular.tools`, `acoular.demo` submodules

### CODE REVIEW

File: .ruff.toml [hunk 1]
```diff
@@ -61,7 +61,7 @@ fixable = ["ALL"]
 unfixable = []
 
 [lint.pydocstyle]
-convention = "google"  # Accepts: "google", "numpy", or "pep257".
```
Comment (artpelling) [inline .ruff.toml:64]: I wondered about that one already. 😸

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -7,11 +7,7 @@ on:
     branches: [ master ] # build docs as additional test on PRs to main
   workflow_dispatch:
     branches: [ '*' ] # allow manual trigger for all branches
-
-concurrency:
-  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
-  cancel-in-progress: true
-
```
Comment (artpelling) [inline .github/workflows/docs.yml:14]: Why did you remove concurrency?

**Review by artpelling — APPROVED** [review]

File: .github/workflows/docs.yml (hunk 2 above)
Comment (adku1173) [reply .github/workflows/docs.yml:14]: ups, this was not intended. Should be fixed now!

**Review by artpelling — APPROVED** [review]
